File: backend/app/api/response.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel


@dataclass
class VideoInfo:
    video_id: UUID
    video_path: str
    user_id: UUID
    user_name: str
    title: str
    description: Optional[str]
    upload_time: datetime
    thumbnail_path: str
    categories: list["Category"]


@dataclass
class Category:
    category_name: str


class BaseResponse(BaseModel):
    message: str = "success"


class UserInfoResponse(BaseResponse):
    account: str
    email: str
    user_name: str


class GetHomeVideoResponse(BaseResponse):
    video_list: list[VideoInfo]


class GetVideoListByTagResponse(BaseResponse):
    video_list: list[VideoInfo]


class GetVideoInfoResponse(BaseResponse):
    video_info: VideoInfo


class GetAllCategoryResponse(BaseResponse):
    categories: list["CategoryForGetAllCategoryResponse"]


@dataclass
class CategoryForGetAllCategoryResponse:
    category_name: str
    category_id: UUID

File: frontend/src/common/response.ts
import { Category, VideoInfo } from "@/lib/redux/features/videoInfoSlice";

export type BaseResponse = {
  message: string;
};

export type GetChannelInfoResponse = {
  message: string;
  account: string;
  email: string;
  user_name: string;
};

export type LogOutResponse = {
  message: string;
};

export type RegisterResponse = {
  message: string;
};

export type UploadVideoApiResponse = {
  message: string;
};

export type GetHomeVideoResponse = {
  video_list: VideoInfo[];
};

export type GetChannelVideoApiResponse = {
  video_list: VideoInfo[];
};

export type GetTagVideoResponse = {
  video_list: VideoInfo[];
};

export type GetVideoInfoResponse = {
  video_info: VideoInfo;
};

export type CategoryForGetAllCategoryResponse = {
  categories: Category[];
};

export type GetCommentsResponse = {
  comments: Comments[];
  total: number;
};

export type Comments = {
  id: string;
  video_id: string;
  account: string;
  comment_message: string;
  comment_time: Date;
  replies?: Reply[];
};

export type Reply = {
  id: string;
  account: string;
  comment_message: string;
  comment_time: Date;
};
